x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```diff
@@ -0,0 +1,396 @@
+import org.elasticsearch.index.mapper.TextSearchInfo;
+import org.elasticsearch.index.mapper.ValueFetcher;
+import org.elasticsearch.index.mapper.vectors.DenseVectorFieldMapper;
+import org.elasticsearch.index.mapper.vectors.SparseVectorFieldMapper;
+import org.elasticsearch.index.query.SearchExecutionContext;
+import org.elasticsearch.inference.SimilarityMeasure;
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentLocation;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xcontent.XContentParserConfiguration;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+
+import static org.elasticsearch.xpack.inference.mapper.SemanticTextField.CHUNKED_EMBEDDINGS_FIELD;
+import static org.elasticsearch.xpack.inference.mapper.SemanticTextField.CHUNKED_TEXT_FIELD;
+import static org.elasticsearch.xpack.inference.mapper.SemanticTextField.CHUNKS_FIELD;
+import static org.elasticsearch.xpack.inference.mapper.SemanticTextField.INFERENCE_FIELD;
+import static org.elasticsearch.xpack.inference.mapper.SemanticTextField.INFERENCE_ID_FIELD;
+import static org.elasticsearch.xpack.inference.mapper.SemanticTextField.getOriginalTextFieldName;
+
+/**
+ * A {@link FieldMapper} for semantic text fields.
+ */
+public class SemanticTextFieldMapper extends FieldMapper implements InferenceFieldMapper {
+    public static final String CONTENT_TYPE = "semantic_text";
+
+    private static final Logger logger = LogManager.getLogger(SemanticTextFieldMapper.class);
+
+    public static final TypeParser PARSER = new TypeParser(
+        (n, c) -> new Builder(n, c.indexVersionCreated()),
+        notInMultiFields(CONTENT_TYPE)
+    );
+
+    public static class Builder extends FieldMapper.Builder {
+        private final IndexVersion indexVersionCreated;
+
+        private final Parameter<String> inferenceId = Parameter.stringParam(
+            "inference_id",
+            false,
+            mapper -> ((SemanticTextFieldType) mapper.fieldType()).inferenceId,
+            null
+        ).addValidator(v -> {
+            if (Strings.isEmpty(v)) {
+                throw new IllegalArgumentException("field [inference_id] must be specified");
+            }
+        });
+
+        private final Parameter<SemanticTextField.ModelSettings> modelSettings = new Parameter<>(
+            "model_settings",
+            true,
+            () -> null,
+            (n, c, o) -> SemanticTextField.parseModelSettingsFromMap(o),
+            mapper -> ((SemanticTextFieldType) mapper.fieldType()).modelSettings,
+            XContentBuilder::field,
+            (m) -> m == null ? "null" : Strings.toString(m)
```

Comment:
I think we were trying to avoid defining `toString()` here but reusing the `Strings.toString()` and using `toXContent()` to render the conflicts message. 

I don't see that we're doing that anywhere else, so I'm implementing `toString()` on the `ModelSettings` class and updating the tests accordingly.

@jimczi , LMK if I'm missing anything here!

Reply:
We can define `toString` but this part is about serialising the object in case of conflict so we need to handle `null` values. To simplify things we can define toString on the `ModelSettings` class (using Strings.toString(this) should be fine) but we have to use `Objects.toString(m)` to handle null values. 
Not that `Objects#toString` handles `null` values by returning "null" so it would be equivalent to what we're doing here.